code/_onclick/hud/rendering/plane_master.dm

```diff
@@ -577,6 +577,17 @@ INITIALIZE_IMMEDIATE(/atom/movable/screen/plane_master)
 	plane = GHOST_PLANE
 	render_relay_planes = list(RENDER_PLANE_NON_GAME)
 
+/atom/movable/screen/plane_master/ghost/show_to(mob/living/mymob)
+	. = ..()
+	if(!.)
+		return
+
+	var/opacity = MOUSE_OPACITY_OPAQUE
+	// If the mob is not dead, don't let them click on ghosts
+	if(istype(mymob) && mymob.stat != DEAD)
+		opacity = MOUSE_OPACITY_TRANSPARENT
+	mouse_opacity = opacity
```

Comment:
implicit istype sucks, also I think the else pattern is better here
```suggestion
	// If the mob is not dead, don't let them click on ghosts
	if(isliving(mymob) && mymob.stat != DEAD)
		mouse_opacity = MOUSE_OPACITY_TRANSPARENT
	else
		mouse_opacity = MOUSE_OPACITY_OPAQUE
```
Of note, the != dead thing will not work for you. since show_to is only called when the plane master is displayed to the mob.
If you want to do something like this you'd want to do this in Initialize (we feed in the hud that the plane master is owned by, can pull the mob from that) and register for stat change off the mob. then you have a proc linked to that signal that determines the opacity we should use. like a refresh sorta thing